### PR TITLE
Fix #574 (deadlocks in player.c)

### LINKED ISSUE #574: Deadlocks in player.c
We have a serious problem with deadlocks in player.c. When the player asks the UI thread for the next track, it will block until the UI thread responds and continues to hold one of the player locks while it waits. If the UI thread at the same time tries to acquire one of the player locks for some reason (e.g. to change the track manually) we have a deadlock.

I've discovered this while debugging another problem that caused `cmus_get_next` to be called in rapid succession. Luckily this doesn't seem to happen that often normally.

Right now I see two ways to fix this:
- Make `cmus_get_next` async: This is hard because the player code is very complicated and uses a completely synchronous design.
- Release the locks while waiting in `cmus_get_next`: This is easier but still quite complicated. It is unclear which invariants could be violated if we temporarily release the lock.

### PR DESCRIPTION
**Pull request by mahkoh**
Tl;dr: The UI thread no longer waits for the player to perform its requests (player_next, etc.) Instead it simply schedules the commands and returns immediately. This allows the UI thread to reply to `cmus_get_next` requests that occur while the player holds its locks.

Fixes #574.

### CODE REVIEW

**Comment by mahkoh** [comment]
I'm still figuring out some places where the code assumed that the functions were synchronous.

**Comment by flyingmutant** [comment]
Thanks a lot for the PR! The review can take some time though, sorry about that.

**Comment by mahkoh** [comment]
Don't worry about it. It's not done yet though so don't bother with the review yet.

**Comment by tagniam** [comment]
@mahkoh Will the delegate functions in 0654847 allow any command to be run asynchronously? They look like they would help me in addressing the issues of #675.

**Comment by Wang-Yue** [comment]
This have been blocking the release for so long.

**Comment by flyingmutant** [comment]
True.

**Comment by Wang-Yue** [comment]
Please revert this commit. This breaks CoreAudio's mixer initializer.
